app/pipeline.h

```diff
@@ -16,6 +16,12 @@ namespace MouseTrack {
 /// - It knows how to acquire new frames
 /// - It knows how to push the frames through the pipeline steps
 /// - It informs other objects about (inbetween-)results via an observer pattern (PipelineObserver)
+///
+/// The pipeline runs one or multiple worker threads to process the frame stream concurrently.
+/// Start the processing with `start()`. If you want to terminate the pipeline early, call `stop()`,
+/// a signal will be sent to the pipeline, it should stop as soon as possible.
+/// Always call `join` before terminating, it will make sure that all worker threads terminated and delivered their data.
```

Comment:
Still not so sure what you mean. I guess the explanation makes sense. But the verb "join" doesn't seem to correspond to the explanation. Maybe I'm not understanding the concept itself.

Reply:
Don't know, I'm not a native speaker. The terminology comes from the world of concurrency, where join is the standard term to merge two execution paths into one (see std::thread::join() as an example)